lemminx-maven/src/main/java/org/eclipse/lemminx/maven/DOMUtils.java

```diff
@@ -36,6 +39,26 @@ public static DOMNode findClosestParentNode(final IPositionRequest request, fina
 		}
 		return null;
 	}
+	
+	// TODO: use optional?
+	public static DOMNode findNodeInDocument(final DOMDocument document, final String localName) {
```

Comment:
@angelozerr is there a better way to do this? Maybe something in the Lemminx API that I missed?

The function just searches all nodes in a document for a given node.

Reply:
No LemMinx doesn't provides this kind of search because it's too specifiic. You could use XPathManager for that, but as it's a simple search I will do that like you have done.

I will rename your method with `findNodeByLocalName`. What are you using a copy of nodes? Looping for nodes directly from DOM document should work too, no?

Reply:
Good suggestion on the rename - going to do that :)

The reason I copy the nodes is so that the `while` condition evaluates to false until all nodes in the document have been checked. Future child nodes that get encountered will be added to the list of nodes to iterrate through. My technique might be redundant however :P Maybe there's a better way to do this.